lib/ui.lib.php

```diff
@@ -713,7 +713,7 @@ function toggle_visible($element)
 function display_notification($message, $timeout = 5000)
 {
     echo "<script type='text/javascript'>";
-    echo "displayNotification('" . addslashes(json_encode($message)) . "', " . $timeout . ");";
+    echo "displayNotification('" . addslashes(json_encode($message, JSON_UNESCAPED_UNICODE)) . "', " . $timeout . ");";
```

Comment:
Support for PHP <5.4 has already been dropped, as some parts of the code use the short array syntax (`[]`). Even Debian oldstable has PHP 5.4.

About where to escape/unescape, we indeed have two solutions. I don't know which one is the best.